src/js/new-visualization-modal.js

```diff
@@ -54,7 +47,54 @@ export default function initModal() {
   // add SQL query to the modal
   document.getElementById('data-query').innerHTML = dataQuery;
 
+  hideAllWarnings();
+
   // wire up event listeners
   $('#new-vis-button').click(onNewVisSubmit);
   $('#copy-query-button').click(onCopyQueryClick);
 }
+
+const csvStringToArray = (strData) => {
+  const objPattern = new RegExp('(\\,|\\r?\\n|\\r|^)(?:"([^"]*(?:""[^"]*)*)"|([^\\,\\r\\n]*))', 'gi');
+  // Create an array to hold our individual pattern
+  // matching groups.
+  let arrMatches = null,
+    // Create an array to hold our data. Give the array
+    // a default empty first row.
+    arrData = [[]];
+  // Keep looping over the regular expression matches
+  // until we can no longer find a match.
+  while ((arrMatches = objPattern.exec(strData))) {
+    if (arrMatches[1].length && arrMatches[1] !== '\\') arrData.push([]);
+    arrData[arrData.length - 1].push(arrMatches[2] ? arrMatches[2].replace(new RegExp('""', 'g'), '"') : arrMatches[3]);
+  }
+  return arrData;
+};
+
+function readSingleFile(evt) {
```

Comment:
Please comment your function definitions

Reply:
Self explanatory. Any further documentation is noise than helpful